### DISCUSSION TITLE
posts/Go/live_reload_with_air/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#339] [Announcements]
https://www.liwenzhou.com/posts/Go/live_reload_with_air/ 

 李文周的Blog live reload 热加载 实时加载 实时重启 live-reload gin

**Comment by chjiyun** [comment]
跟官方的默认配置一样，air启动还是报错
“windows找不到文件 'APP_ENV=dev' ” 
go 1.15    air  1.12.4

-----------------
去掉APP_ENV=dev APP_USER=air 后启动会打开cmd窗口，有点别扭

**Comment by joewongex** [comment]
> @chjiyun
> 跟官方的默认配置一样，air启动还是报错
> “windows找不到文件 'APP_ENV=dev' ” 
> go 1.15    air  1.12.4
> 
> -----------------
> 去掉APP_ENV=dev APP_USER=air 后启动会打开cmd窗口，有点别扭

windows下把full_bin配置项注释掉就不会弹cmd窗口了

**Comment by Xzhiming** [comment]
怎么停止跑起来的程序啊    Ctrl+c  Air 程序停了   删除tmp下的文件失败  然后里边的两个业务程序还在跑着呢

**Comment by Alancere** [comment]
[GIN-debug] [ERROR] listen tcp :8000: bind: Only one usage of each socket address (protocol/network address/port
) is normally permitted.
为什么启动air后，修改了监听的文件，出现以前的应用程序没有关闭掉啊。

**Comment by chjiyun** [comment]
> @804873052
> [GIN-debug] [ERROR] listen tcp :8000: bind: Only one usage of each socket address (protocol/network address/port
> ) is normally permitted.
> 为什么启动air后，修改了监听的文件，出现以前的应用程序没有关闭掉啊。

跟你一样的问题，升级go1.16 后又出现了，总是出现 tmp/main.exe~ 文件 有进程占用了端口

**Comment by jqcchen** [comment]
@chjiyun
@804873052
@Xzhiming  

 
我也遇到端口占用的问题，windows此配置要注释，不然热更新后会报端口占用
full_bin =“”

**Comment by dd360-dev** [comment]
windows怎么安装环境变量啊，一直报错找不到air命令

**Comment by jeongseup** [comment]
Can I ask you what kind of vscode extensions in this post?

**Comment by xiayin1992728** [comment]
windows 下面安装：
```
（1）go get -u github.com/cosmtrek/air

（2）在 air 包目录中运行 : go build 生成 air.exe

（3）把 air.exe 剪切到 GO_PATH 目录,或者设置环境变量

（4）新建 .air.conf 文件，并复制 air 包里面的 air_example.toml 内容到 .air.conf 修改配置

（5）在项目目录中运行： air -c .air.conf

**Comment by joewongex** [comment]
你好，你的邮件我已收到
